### ISSUE TITLE
More Indicator sets

### ISSUE THREAD

**Issue by mfalt** [kul-forbes/ProximalOperators.jl#2]
We should try to add all the indicator sets from [MathProgBase](http://mathprogbasejl.readthedocs.io/en/latest/conic.html). The current status is
- [x] :Free (implemented by `IndBox(-Inf, Inf)`)
- [x] :Zero (implemented by `IndBox(0, 0)`)
- [x] :NonNeg (implemented by `IndBox(0, Inf)`)
- [x] :NonPos (implemented by `IndBox(-Inf, 0)`)
- [x] :SOC - `IndSOC`
- [x] :SOCRotated (Should not be tricky)
- [x] :SDP - `IndPSD` (Can we have a compatible version for the vector form? Should we change the name to IndSDP?)
- [x] :ExpPrimal
- [x] :ExpDual (be be implemented using ExpPrimal)

It is possible that we could get slightly better performance by special versions of the first 4.
For now we could also simple introduce some aliases for the first 4 and worry about special implementations later.

**Comment by lostella** [comment]
I actually put directly the first two in a more optimized way...

What do you mean by "vector form"? You mean getting Array arguments rather than Hermitian or Symmetric ones?

**Comment by mfalt** [comment]
The variables in MathProgBase corresponding to a SDP cone are represented in a single vector according to `loadproblem!` in the link in my original post. So to go from the vector to a Symmetric one would do something like

``` julia
# Converts a vector representation of a symmetrix matrix.
# The off diagonal elements are assumed to be scaled by a factor of sqrt(2) in the input.
function vec2symmetric(x::AbstractVector)
  y = length(x)
  n = try
       Int(sqrt(1/4+2*y)-1/2)
  catch
       error("Invalid size of Symmetric matrix represented as vector")
  end
  A = zeros(n,n)
  for j = 1:n, i = j:n
    fac = (i==j ? 1.0 : 1.0/sqrt(2))
    A[i,j] = x[k]*fac
    k = k+1
  end
  return Symmetric(A, :L)
end
```

It would be nice if the projection could be done without explicitly constructing this matrix every time prox is called. In particular, there used to be a julia type `SymmetricRFP` and corresponding LAPACK commands to do operations on the vector form. I'm still trying to research if it is coming back or if there is a good alternative.

**Comment by mfalt** [comment]
Great that you added Exp cones, that will be very useful for me! I created a PR  #3 for SDP in packed matrix form, do what you want with it ;). Not sure why it failed in nightly v0.6

**Comment by lostella** [comment]
Last item (:SOCRotated, implemented by `indRotatedSOC`) solved by #13. Closing.

**Comment by mfalt** [comment]
Great job! I'll definitely use this. I'm planning on releasing a package for first order methods that is using this, probably before summer.  It would be nice with some input from you, so you have somewhere we could chat?
